### PR TITLE
[breaking]: use go embed; remove packr; remove helm_template kind; update dependencies

### PR DESCRIPTION
**Pull request by edulop91**
### Summary
There are a bunch of changes here, apologies for the dump in advance:

- Updating all deps.
- terraform core packages, such as plan, are now internal and can't be imported. Removing experimental commands that depended on them.
- go 1.16 stdlib allows for embedding of files, use that instead of packr. One annoying bit is we have to enumerate all embeded files that start with `.` 

### Test Plan
- running in internal repo
- make sure only change in golden test files are:
   - removal of the entropy command
   - removal of kind: helm components

### CODE REVIEW

File: docs/reference/config_file.md [hunk 1]
```diff
@@ -207,7 +207,7 @@ Each workspace within an environment is called a `component`
 In addition to the common workspace arguments, components also support the following arguments:
 
 - `eks` - Object that contains a `cluster_name` key to point terraform to a kubectl context
-- `kind` - String, either `terraform` or `helm_template` (DEPRECATED)
```
Comment (jgadling) [inline docs/reference/config_file.md:210]: is this still deprecated?

File: .goreleaser.yml [hunk 2]
```diff
@@ -1,7 +1,6 @@
 before:
   hooks:
     - make clean
-    - make packr
```
Comment (jgadling) [inline .goreleaser.yml:4]: yasssss!!

**Comment by jgadling** [comment]
do we still need refs to packr in `.godownloader-packr.sh` and `.gitignore`?

File: apply/apply.go [hunk 3]
```diff
@@ -202,8 +215,21 @@ func applyEnvs(fs afero.Fs, p *plan.Plan, envBox *packr.Box, componentBoxes map[
...
+		if d.IsDir() {
+			return nil // skip dirs
+		}
+
+		sourceFile, err := source.Open(path)
+		if err != nil {
+			return errs.WrapInternal(err, "could not read source file")
+		}
+
+		// return source.Walk(func(path string, sourceFile packr.File) error {
```
Comment (edulop91) [inline apply/apply.go]:
```suggestion
```

File: docs/reference/config_file.md (hunk 1 above)
Comment (edulop91) [reply docs/reference/config_file.md:210]: yes (does nothing when empty or kind == terraform, fails when anything else)

File: templates/templates_test.go [hunk 4]
```diff
@@ -41,4 +41,6 @@ func TestOpenTemplate(t *testing.T) {
 			r.NotNil(temp.Templates())
 		})
 	}
+
+	// t.FailNow()
```
Comment (edulop91) [inline templates/templates_test.go:45]:
```suggestion
```

File: templates/templates_test.go [hunk 5]
```diff
@@ -41,4 +41,5 @@ func TestOpenTemplate(t *testing.T) {
 			r.NotNil(temp.Templates())
 		})
 	}
+
```
Comment (cziie) [inline templates/templates_test.go:44]: 🚫 **[golangci]** <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>unnecessary trailing newline (whitespace)

**Review by jgadling — APPROVED** [review]

**Pull request merged**